metecho/api/serializers.py

```diff
@@ -364,23 +364,55 @@ def update(self, instance, validated_data):
         if instance.assigned_dev != validated_data["assigned_dev"]:
             if validated_data.get("should_alert_dev"):
                 self.try_send_assignment_emails(instance, "dev", validated_data, user)
-            orgs = instance.scratchorg_set.active().filter(
-                org_type=SCRATCH_ORG_TYPES.Dev
-            )
+            # We want to consider soft-deleted orgs, too:
+            orgs = instance.scratchorg_set.all().filter(org_type=SCRATCH_ORG_TYPES.Dev)
+            reassigned_org = False
             for org in orgs:
-                org.queue_delete(originating_user_id=originating_user_id)
+                new_user = self._valid_reassign(
+                    "dev", org, validated_data["assigned_dev"]
+                )
+                valid_commit = org.latest_commit == (
+                    instance.commits[0] if instance.commits else instance.origin_sha
+                )
+                if new_user and valid_commit and not reassigned_org:
+                    org.queue_reassign(
+                        new_user=new_user, originating_user_id=originating_user_id
+                    )
+                    reassigned_org = True
```

Comment:
@wlonk Note from today's mtg: We want to add an endpoint solely for the purpose of knowing whether an existing (dev) org can be reassigned from one user to another.

Reply:
This should just be an endpoint that does the same logic as is currently in `TaskSerializer._valid_reassign`, right? Or is there another part to it that I'm forgetting (cc @davisagli )

Reply:
@wlonk I think that's correct, yes. The front end provides an org and a GH user, and the endpoint returns a boolean (effectively).